CMakeLists.txt

```
@@ -63,8 +63,16 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
-
-find_package(Z3 4.8.0)
+set(LATEST_Z3_VERSION "4.8.12")
+if (TESTS AND NOT EMSCRIPTEN)
+  find_package(Z3 ${LATEST_Z3_VERSION} REQUIRED)
+  if (NOT ${Z3_FOUND})
+    message(WARNING "SMTChecker tests require Z3 ${LATEST_Z3_VERSION} for all tests to pass.\
+                   \nuse -DTESTS=OFF, if you want to use older version of z3")
```

I think that the whitespace before `\n` will be included in the message, which could make it wrap in a weird way on a very narrow terminal.

I'd indent and wrap it this way:
```suggestion
    message(
      WARNING
      "SMTChecker tests require Z3 ${LATEST_Z3_VERSION} for all tests to pass.\n\
Build with -DTESTS=OFF if you want to use an older version. \
You can also use -DUSE_Z3=OFF to build without Z3 and then use --no-smt when running tests."
    )
```

---

good suggestion.